starter-code/src/memory.js

```diff
@@ -1,15 +1,45 @@
-// var MemoryGame = function (cards) {
-//   this.cards = cards;
-// };
+var MemoryGame = function (cards) {
+  this.cards = cards;
+  this.pickedCards = [];
+  this.pairsClicked = 0;
+  this.pairsGuessed = 0;
+  this.isMatch = false;
+};
 
-// MemoryGame.prototype.shuffleCard = function (cardsArr) {
+MemoryGame.prototype.shuffleCard = function (cardsArr) {
+ var currentIndex = cardsArr.length, temporaryValue, randomIndex;
 
-// };
+ while (0 !== currentIndex) {
+     randomIndex = Math.floor(Math.random() * currentIndex);
+     currentIndex -= 1
 
-// MemoryGame.prototype.checkIfPair = function (firstCard, secondCard) {
+    temporaryValue = cardsArr[currentIndex];
+    cardsArr[currentIndex] = cardsArr [randomIndex];
+    cardsArr[randomIndex] = temporaryValue
+ }
+ return cardsArr;
+};
 
-// }
+MemoryGame.prototype.checkIfPair = function (firstCard, secondCard) {
+    this.pairsClicked++;
+    if (firstCard === secondCard) {
+        this.pairsGuessed++;
+        this.isMatch = true;
+        return true;
+    } else {
+        return false;
+    }
+}
 
-// MemoryGame.prototype.finished = function () {
+MemoryGame.prototype.finished = function () {
+    var numberOfPairs = this.cards.length / 2;
+    if (this.pairsClicked == 0) {
+        return false
+    }
```

Comment:
```
+    if (this.pairsClicked == 0) {
 +        return false
 +    }
```
This and the last else basically is matching `(this.pairsGuessed !== numberOfPairs)` so we can put only 2 cases. 
```
+     if (this.pairsGuessed === numberOfPairs) {
 +        return true;
 +    } else {
 +        return false;
 +    }
```